src/bottom-sheet.ts

```diff
@@ -28,8 +28,33 @@ type SheetUI = Readonly<{
     content: Array<SheetUIElement>;
 }>;
 
-export const bottomSheet = (payload: SheetUI): Promise<SheetResponse> =>
-    postMessageToNativeApp({type: 'SHEET', payload});
+let sheetLock = false;
+
+type SheetLockedResponse = {action: 'LOCKED'; result: Array<any>};
+
+export const bottomSheet = async (
+    payload: SheetUI,
+): Promise<SheetResponse | SheetLockedResponse> => {
+    if (sheetLock) {
+        return Promise.resolve({action: 'LOCKED', result: []});
+    }
+
+    sheetLock = true;
+    const tid = setTimeout(() => {
+        sheetLock = false;
+    }, 1000);
```

Comment:
maybe 3 seconds or so is better than just a second ?

Reply:
the idea is to have a lock for the time between the user taps and the bottom sheet appears. I think that the bottom sheet should appear in less than 1 second